Kernel/CMakeLists.txt

```diff
@@ -852,8 +852,17 @@ add_dependencies(Kernel generate_EscapeSequenceStateMachine.h generate_version_h
 target_link_libraries(Kernel PUBLIC GenericClangPlugin)
 
 if("${SERENITY_ARCH}" STREQUAL "aarch64")
-    target_link_options(Kernel PRIVATE LINKER:-T ${CMAKE_CURRENT_SOURCE_DIR}/Arch/aarch64/linker.ld -nostdlib LINKER:--no-pie)
-    set_target_properties(Kernel PROPERTIES LINK_DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/Arch/aarch64/linker.ld)
+    add_custom_command(
+        OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/linker.ld
+        COMMAND "${CMAKE_CXX_COMPILER}" ${TARGET_STRING} -E -P -x c -I${CMAKE_CURRENT_SOURCE_DIR}/.. "${CMAKE_CURRENT_SOURCE_DIR}/Arch/aarch64/linker.ld" -o "${CMAKE_CURRENT_BINARY_DIR}/linker.ld"
```

Comment:
The title for this commit should probably be "Kernel/aarch64: Build kernel as position-independent code" instead of the current "Kernel/aarch64: Make the kernel bootable via the EFI Prekernel"?

Mention in commit message that the ld script is now getting preprocessed, and that we no longer build with --no-pie.